File: Interview Prep/Matrix To String/MatrixToStringConversion.java
import java.util.Scanner;

public class MatrixToStringConversion {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int t = scanner.nextInt();
    while (t-- > 0) {
      int M = scanner.nextInt();
      int N = scanner.nextInt();
      int[][] matrix = new int[M][N];
      for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
          matrix[i][j] = scanner.nextInt();
        }
      }
      String[] result = matrixToStringConversion(matrix, M, N);
      for (String row : result) {
        System.out.println(row);
      }
    }
    scanner.close();
  }

  public static String[] matrixToStringConversion(int[][] matrix, int M, int N) {
    String[] result = new String[M];
    for (int i = 0; i < M; i++) {
      StringBuilder rowString = new StringBuilder();
      for (int j = 0; j < N; j++) {
        if (matrix[i][j] == 0) {
          rowString.append('.');
        } else {
          rowString.append((char) (matrix[i][j] + 64));
        }
      }
      result[i] = rowString.toString();
    }
    return result;
  }
}

File: Interview Prep/Matrix To String/MatrixToStringConversion.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

vector<string> matrixToStringConversion(const vector<vector<int>>& matrix,
                                        int M, int N) {
  vector<string> result(M);
  for (int i = 0; i < M; ++i) {
    string rowString;
    for (int j = 0; j < N; ++j) {
      if (matrix[i][j] == 0) {
        rowString += '.';
      } else {
        rowString += (char)(matrix[i][j] + 64);
      }
    }
    result[i] = rowString;
  }
  return result;
}

int main() {
  int t;
  cin >> t;
  while (t-- > 0) {
    int M, N;
    cin >> M >> N;
    vector<vector<int>> matrix(M, vector<int>(N));
    for (int i = 0; i < M; ++i) {
      for (int j = 0; j < N; ++j) {
        cin >> matrix[i][j];
      }
    }
    vector<string> result = matrixToStringConversion(matrix, M, N);
    for (const string& row : result) {
      cout << row << endl;
    }
  }
  return 0;
}
